### PR TITLE
docs/packages: Set up repo side of i18n process

### PR DESCRIPTION
**Pull request by kaechele**
This is to address #1593.
With this in place we can maintain translations for docs, internal packages and the site example in Weblate.
I have set up an Instance at https://translate.kaechele.ca.

It does however require that someone runs `make update-po` in docs/ before each commit of changes to the documentation. For this the person running the command must have *sphinx* and *sphinx-intl* installed on their machine.

In a next step we could set up GitHub webhooks to alert my Weblate instance of any changes to the repo.

In the last and final step, as soon as we are confident it works as intended, we could allow the Weblate instance to push the updated translations back to the repository automatically.

### CODE REVIEW

**Comment by rubo77** [comment]
Shouldn't long lines be broken at 70 characters like everywhere?

**Comment by bobcanthelpyou** [comment]
@rubo77 is a style guide for Gluon code & docs available? I don't know any place where a max 70 characters rule is applied. I know the Development Guidelines with "rules", only. It is very straightforward. If there are rules missing add these to the docs, please. https://gluon.readthedocs.io/en/latest/dev/basics.html#development-guidelines

**Comment by kaechele** [comment]
All files in this PR except for docs/conf.py and docs/Makefile are automatically generated. I'm not aware of a setting to have the generator put in line breaks.
Also the generated files are meant to be consumed by Weblate and not to be edited by the user directly, although that is still possible.

**Comment by NeoRaider** [comment]
If anything, a 80 or 120 character rule is common, but none is enforced at this point. As @kaechele points out, these files are generated automatically, so no style rules apply to them anyways.

**Comment by CodeFetch** [comment]
@kaechele Can we start to translate using this Weblate instance already?

**Comment by kaechele** [comment]
Yes. I can merge all changes onto the actual gluon repo once we switch over to that.

File: docs/releases/v2018.2.1.rst [hunk 1]
```diff
@@ -34,6 +34,10 @@ ramips-mt76x8
   - Archer C50 (v3, v4) [#noibss]_
   - TL-WR841N (v13) [#noibss]_
 
+.. [#noibss]
+  Device or target does not support AP+IBSS mode: This device or target will not be built
+  when *GLUON_WLAN_MESH* is set to ``ibss``.
+
```
Comment (rotanid) [inline docs/releases/v2018.2.1.rst]: i think this is out of scope of this PR and should be done separately/beforehand for all affected release notes

Comment (kaechele) [reply docs/releases/v2018.2.1.rst]: It is in scope so far as this generates an error when generating the gettext po files. So while not directly in scope it fixes errors that are made visible by this change. But I'm happy to pull this out of the PR.

**Comment by kaechele** [comment]
Let me know when this is ready to go in. Because there have been changes to files that warrant a re-run of the po-file generation. This is probably something we'd want to do before merging this.

**Comment by kaechele** [comment]
Here is the documentation on configuring GitHub and Weblate for webhooks so that Weblate is notified on updates to the translation files: https://docs.weblate.org/en/latest/admin/continuous.html
I'm currently looking into [using a CI](https://developer.github.com/v3/guides/building-a-ci-server/) server to automatically regenerate the translation files upon changes to the docs.

**Comment by CodeFetch** [comment]
@mweinelt This should work like the [phpmyadmin documentation](https://docs.phpmyadmin.net/de/latest/).
You can select the language where you can select the version.

**Comment by kaechele** [comment]
> How were all those files in `docs/locales/` and `docs/_build/gettext` created? Automatically by running `make -C docs update-po`? What was created manually and needs to be reviewed then?

The following files contain manual changes or were created manually and require review:

- docs/.gitignore
- docs/Makefile
- docs/releases/v2018.2.1.rst

All other files (specifically those with a *.po* and *.pot* suffix) are automatically generated by running `make -C docs update-po`
I guess I should mention that the update-po directive requires [sphinx-intl](https://pypi.org/project/sphinx-intl/) to be installed. sphinx-intl is part of the regular package repository of all major Linux distributions.

**Comment by kaechele** [comment]
I just noticed that I'll have to do some more testing here since every run of `make -C docs update-po` generates new hashes within the files and thus changes every single file every time.

**Comment by CodeFetch** [comment]
What's the problem with this PR? I think that the hashes within the files change is nothing critical...

**Comment by christf** [comment]
@kaechele The gluon team had a PR review session a few days ago. The language of the project is english. All documentation and communication should thus be in english. We welcome internationalization because we feel that there should be as little barries for understanding as possible. At the same time we are concerned that we cannot keep the translations maintained and by simply merging the PR as-is could create a situation where we are unable to cope with creating the original documentation and its translations.

Creating a translated documentation with missing text bodies is a concern too. We would like to avoid a situation where the translated documentation is viewed as the ugly step-sibling, at the same time we would like to spend what little ressources we have in the core team on developing gluon furter.

We wamt to follow this strategy:
 
- in gluon, integrate the required framework to support translations to many languages
- find a way to maintain the actual translations outside the main gluon repository allwoing for different release cycles between translations and gluon itself.
- Link translations from the original documentation to have a good integration.

This means this PR cannot be merged as-is and I apologize for this decision taking longer than it should have. Would you like to work with us to establish the structure as discussed above?

**Comment by kaechele** [comment]
I have no emotions for this. I was just trying to help other people doing translations if they wished. I can look whether Weblate can support what you proposed, if anyone is interested but unable themselves to set up and run Weblate.
I myself don't have an interest in translated docs but am happy to help those who are.

**Comment by CodeFetch** [comment]
@christf Have you tested Weblate? It's exactly what you proposed...

**Comment by christf** [comment]
I have not been aware of weblate until you mentioned it. There is a free service for "free projects". It could be interesting. Let's discuss this on Review Day.

**Comment by CodeFetch** [comment]
@kaechele Your Weblate instance is down. We could move it to the Freifunk Hannover infrastructure for long-term support if you like.

**Comment by mweinelt** [comment]
As it looks this is not going to happen. If noone disagrees I'll close this pull request soon.

**Comment by kaechele** [comment]
Yes, please close for now. I don't have time to work on this right now. After the initial setup maintaining the instance should be easy but the initial work would probably have to be reworked.
